test/bigml/test/api/source.clj

```diff
@@ -8,11 +8,13 @@
   (:use clojure.test))
 
 (defn- create-and-test [artifact & params]
-  (let [initial (api/get-final (apply source/create artifact params))
+  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
```

Comment:
Thanks, @jimshur, I see what you mean.

I have pushed a fix for that. I hope it is sufficiently idiomatic:

     (let [{:keys [username api_key dev_mode]} params
            sticky-pars [:username username :api_key api_key :dev_mode dev_mode]


> On 14 Jan 2016, at 12:11, jimshur <notifications@github.com> wrote:
> 
> In test/bigml/test/api/source.clj:
> 
> > @@ -8,11 +8,13 @@
> >    (:use clojure.test))
> >  
> >  (defn- create-and-test [artifact & params]
> > -  (let [initial (api/get-final (apply source/create artifact params))
> > +  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
> 
> @sdesimone I think your understandable confusion is with the :as directive, which in fact assigns the entire expression being deconstructed, not just the ones listed for the keys. Put (let [{:keys [a b] :as p} {:a 1 :b 2 :c 3}] p) in the repl and see.
> 
> —
> Reply to this email directly or view it on GitHub.
>